src/test/utils.test.ts

```diff
@@ -286,4 +286,52 @@ describe('Utils', () => {
 			should.equal('key @ name', utils.generateReferenceValueForLabel(undefined, new ConfigMapRef(new KeyRef('name', 'key'))));
 		});
 	});
+
+	context('Load/Save of a Secrets file', () => {
+
+		let dummyNonSecretFile: string;
+
+		before( () => {
+			dummyNonSecretFile = path.resolve(__dirname, '../../testFixture', `dummy-secret.yaml`);
+			fs.writeFileSync(dummyNonSecretFile, 'test');
+		});
+
+		after( () => {
+			fs.unlinkSync(dummyNonSecretFile);
+		});
+
+		it('should match the secrets model contents between a save and reload to/from a secret file', async() => {
+			const name: string = 'mysecret';
+			const fpOrig: string = path.resolve(__dirname, '../../testFixture', `${name}.yaml`);
+			const fpTest: string = path.resolve(__dirname, '../../testFixture', `${name}2.yaml`);
+			const yamlDoc:SecretConfig = await utils.loadSecretsFromFile(fpOrig);
+			should.exist(yamlDoc);
+			await utils.saveSecretsToFile(yamlDoc, fpTest);
+			should.exist(utils.validateFileNotExisting(name));
+			const yamlDoc2:SecretConfig = await utils.loadSecretsFromFile(fpTest);
+			should.exist(yamlDoc2);
+			yamlDoc.should.deep.equal(yamlDoc2);
+			fs.unlinkSync(fpTest);
```

Comment:
it is better to take of deletion in afterEach method. it ensures that even if the test is failing, there is no file that is left in the workspace and that can impact other tests. it helps to provide a more accurate vision on which tests are failing when there is a regression.

Reply:
yeah. agreed.